### PR TITLE
VZ-2390.  MC Test Suite - Perform managed cluster and admin cluster install on OKE

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Finish up support in MC AT pipeline for OKE clusters
- Fix OCI DNS install setup, add OKE MagicDNS configuration stage
- Fix conditionals on KinD and OKE/MagicDNS stages
- Consolidate installer configuration into a common function
- Cleanup and refactor large script blocks to separate functions
- Fix up cluster configuration stage names, comment helper fns
- Handle special cases around envName for xip.io vs OCI DNS

Fixes VZ-2390.

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by markxnelson — APPROVED** [review]

File: ci/multicluster/Jenkinsfile [hunk 1]
```diff
@@ -108,14 +110,9 @@ pipeline {
         OCI_CLI_SUPPRESS_FILE_PERMISSIONS_WARNING = 'True'
         INSTALL_CONFIG_FILE_OCIDNS = "${GO_REPO_PATH}/verrazzano/tests/e2e/config/scripts/install-verrazzano-ocidns.yaml"
         INSTALL_CONFIG_FILE_XIPIO = "${GO_REPO_PATH}/verrazzano/tests/e2e/config/scripts/install-verrazzano-xipio.yaml"
-        OCI_DNS_ZONE_SUFFIX=credentials('oci-dns-zone')
-        OCI_DNS_ZONE_NAME="z${zoneId}." + "${OCI_DNS_ZONE_SUFFIX}"
-        VZ_ENVIRONMENT_NAME = "${params.TEST_ENV == 'ocidns_oke' ? 'b' + env.BUILD_NUMBER : 'default'}"
+        OCI_DNS_ZONE_NAME="z${zoneId}.v8o.oracledx.com"
```
Comment (markxnelson) [inline ci/multicluster/Jenkinsfile:113]: use the variable to avoid checking in actual system names

File: ci/multicluster/Jenkinsfile [hunk 2]
```diff
@@ -541,6 +509,93 @@ def runGinkgoRandomize(testSuitePath) {
...
+        int clusterCount = params.TOTAL_CLUSTERS.toInteger()
+        def envName = "default"
+        for(int count=1; count<=clusterCount; count++) {
+            def destinationPath = "${env.KUBECONFIG_DIR}/${count}/${installerFileName}"
+            def installProfile = env.MANAGED_CLUSTER_PROFILE
+            if (count == 1) {
+                installProfile = env.ADMIN_CLUSTER_PROFILE
+            }
+            if (params.TEST_ENV == 'ocidns_oke') {
+                // Installs using OCI DNS requires a unique domain per cluster, so to distinguish
```
Comment (markxnelson) [inline ci/multicluster/Jenkinsfile:524]: require, no 's'

File: ci/multicluster/Jenkinsfile (hunk 1 above)
Comment (michael-cico) [reply ci/multicluster/Jenkinsfile]: Updated

File: ci/multicluster/Jenkinsfile (hunk 2 above)
Comment (michael-cico) [reply ci/multicluster/Jenkinsfile]: fixed

**Pull request merged**